base/src/main/java/org/aya/resolve/visitor/StmtShallowResolver.java

```diff
@@ -36,6 +37,13 @@ public void resolveStmt(@NotNull Stmt stmt, @NotNull ModuleContext context) {
     switch (stmt) {
       case Decl decl -> resolveDecl(decl, context);
       case Command.Module mod -> {
+        var wholeModeName = context.moduleName().appended(mod.name());
+        // Is there a module with path {context.moduleName}::{mod.name} ?
+        if (loader.exists(wholeModeName)) {
+          // 🥲
+          context.reportAndThrow(new NameProblem.DuplicateModNameError(ModulePath.qualified(wholeModeName), mod.sourcePos()));
```

Comment:
It is good to have a new problem saying something like `The inner module ${wholeModName} clashes with some file level module ${wholeModName.replace("::", "/")}.aya`, which looks more friendly.